app/gui2/src/components/MarkdownEditor/ImagePlugin/index.ts

```diff
@@ -8,18 +8,16 @@ import type { TextMatchTransformer } from '@lexical/markdown'
 import type { LexicalEditor } from 'lexical'
 import { assertDefined } from 'shared/util/assert'
 
-export const DEFAULT_ALT_TEXT = 'Image'
-
-const IMAGE: TextMatchTransformer = {
+export const IMAGE: TextMatchTransformer = {
   dependencies: [ImageNode],
   export: (node) => {
     if (!$isImageNode(node)) {
       return null
     }
     return `![${node.getAltText()}](${node.getSrc()})`
   },
-  importRegExp: /!\[([^[]+)]\(([^()\n]+)\)/,
-  regExp: /!\[([^[]+)]\(([^()\n]+)\)$/,
+  importRegExp: /!\[([^\]]*)]\(([^()\n]+)\)/,
+  regExp: /!\[([^\]]*)]\(([^()\n]+)\)$/,
```

Comment:
Any hints in a comment? These regexes are really hard to comprehend.

Reply:
Well, as usual, it’s easier to rewrite the regexp from scratch than to edit the existing one. Here the thick was in replacing `+` with `*`, but I also did replace `[` with `]` in excluded symbols, because it feels more natural. (the idea is "we take `[`, then any symbol that is not `]`, then `]`)
I don’t think the comment would help in the future, honestly.